sdk/template/Azure.Template/src/Generated/TemplateClient.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Template
+{
+    /// <summary> The Template service client. </summary>
+    public partial class TemplateClient
+    {
+        private static readonly string[] AuthorizationScopes = new string[] { "https://api.adu.microsoft.com/.default" };
+        private readonly TokenCredential _tokenCredential;
+        private readonly HttpPipeline _pipeline;
+        private readonly string _vaultBaseUrl;
+        private readonly string _apiVersion;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal ClientDiagnostics ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of TemplateClient for mocking. </summary>
+        protected TemplateClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of TemplateClient. </summary>
+        /// <param name="vaultBaseUrl"> The vault name, for example https://myvault.vault.azure.net. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="vaultBaseUrl"/> or <paramref name="credential"/> is null. </exception>
+        public TemplateClient(string vaultBaseUrl, TokenCredential credential, TemplateClientOptions options = null)
```

Comment:
This constructor doesn't follow the Azure SDK guidelines -- it should take a Uri for the endpoint, an have no default parameters in the constructor.  We should regenerate once https://github.com/azure/autorest.csharp/issues/2105 is resolved.

Reply:
This is generated by current autorest.csharp. When the autorest.csharp upgrade, the file should be re-generated.